addon/mixins/css.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Mixin for providing a `css` property on components
+ * The `css` property will be the introspected name of the component
+ */
+import Ember from 'ember'
+const {Mixin} = Ember
+import {PropTypes} from 'ember-prop-types'
+
+export default Mixin.create({
+
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  classNameBindings: ['css'],
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    css: PropTypes.string
+  },
+
+  /**
+   * @returns {Object} the default properties for this mixin
+   */
+  getDefaultProps () {
+    return {
+      css: this.getComponentName()
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  // == Functions =============================================================
+
+  /**
+   * @returns {String} the name of the component, parsed out of the toString() result
+   */
+  getComponentName () {
+    return this.toString().replace(/^.+:(.+)::.+$/, '$1')
```

Comment:
Do we want to mark this as using a private function?  I don't know how we want to track those kinds of things